### DISCUSSION TITLE
Action enhancements

### DISCUSSION THREAD

**Discussion by rwilliam99** [nhielost/obs-midi-mg#94] [Ideas]
It would be really useful to be able to assign a key press as an action (hotkeys).  It would greatly extend the functionality of this tool.  I've been using a much more cryptic way of doing this and I'd like to convert to this tool which is WAY more maintainable and understandable.

Thanks for your work on this!

**Comment by phantom-GH** [comment]
Almost a year later but I second this! I have been having a tough time trying to work around this. If you're still around could you tell me what your method is?

**Reply by rwilliam99** [reply]
I've made a lot of changes in the last year and I'm not positive how I was achieving it before, but I think I was invoking hotkeys using a piece of software called Bome MIDI Translator Pro.  I believe this comment was hoping that if Hotkeys were added as an action I could completely replace the Bome software.  Using it worked well, but it was rather cryptic and unintuitive for what I was using it for.  Granted, using MIDI events to trigger OBS isn't exactly using it as designed either but it has worked really well for me.  I believe I was able to re-think how I was doing things and be able to use obs-midi-mg without invoking hotkeys so I could convert to it.

I've switched to using TouchPortal to control OBS and it is working well for me.  Another idea/option is to create a .exe using AutoIt software - it is a scripting language that is pretty rich and you can compile a script to an .exe file to be able to run it without the scripting engine.

Hope that helps a bit - good luck!
